FirebaseAdmin/FirebaseAdmin/Auth/TenantAwareFirebaseAuth.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018, Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using System;
+
+namespace FirebaseAdmin.Auth
+{
+    /// <summary>
+    /// The tenant aware <see cref="FirebaseAuth"/> class. You can
+    /// get an instance of this class via <see cref="TenantManager.AuthForTenant(string)">FirebaseAuth.DefaultInstance.TenantManager.AuthForTenant(tenantId)</see>.
+    /// </summary>
+    public sealed class TenantAwareFirebaseAuth : AbstractFirebaseAuth
+    {
+        internal TenantAwareFirebaseAuth(FirebaseAuthArgs args)
```

Comment:
This constructor should accept a tenant ID as an argument. Tenant ID only makes sense at this level. AbstractFirebaseAuth should be completely unaware of all tenant-scoped operations.